databuilder/databuilder/extractor/query/query_metadata_base.py

```diff
@@ -0,0 +1,354 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from abc import abstractmethod
+from typing import (
+    Any, Dict, Iterator, List, Optional, Union,
+)
+
+from pyhocon import ConfigTree
+
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.models.query import (
+    QueryExecutionsMetadata, QueryJoinMetadata, QueryMetadata, QueryWhereMetadata,
+)
+from databuilder.models.table_metadata import ColumnMetadata, TableMetadata
+from databuilder.models.user import User as UserMetadata
+
+LOGGER = logging.getLogger(__name__)
+
+
+class QueryMetadataBaseExtractor(Extractor):
+    """
+    Provides common funcitons used by query metadata extractors including:
+        - Converting a list of Stemma SQL tables to QueryMetadata,
+        - Converting stemma where cluases to QueryWhereMetadata
+        - Converting stemma joins cluases to QueryJoinMetadata
+    Implementations need to provide the structure that iterates over records,
+    creates the Stemma SQL objects and then passses them to the corresponding
```

Comment:
same

Reply:
Ahh, yep, thanks for catching this. All will be removed in the next commit.